addons/account_tax_python/tests/test_taxes_computation.py

```diff
@@ -0,0 +1,103 @@
+    def _prepare_test(self, formula, *args, **kwargs):
+        tax = self.python_tax(formula, **kwargs.pop('tax_values', {}))
+        product_values = kwargs.pop('product_values', None)
+        if product_values:
+            product = self.env['product.product'].create({
+                'name': "_prepare_test",
+                **product_values,
+            })
+            kwargs.setdefault('evaluation_context_kwargs', {})['product'] = product
+
+        return self._prepare_taxes_computation_test(tax, *args, **kwargs)
+
+    def test_formula(self):
+        tests = [
+            self._prepare_test(
+                "max(quantity * price_unit * 0.21, quantity * 4.17)",
+                130.0,
+                {
+                    'total_included': 157.3,
+                    'total_excluded': 130.0,
+                    'tax_values_list': (
+                        (130.0, 27.3),
+                    ),
+                },
+            ),
+            self._prepare_test(
+                "max(quantity * price_unit * 0.21, quantity * 4.17)",
+                130.0,
+                {
+                    'total_included': 130.0,
+                    'total_excluded': 102.7,
+                    'tax_values_list': (
+                        (102.7, 27.3),
+                    ),
+                },
+                tax_values={'price_include': True},
+            ),
+            self._prepare_test(
+                "product.volume * quantity * 0.35",
+                100.0,
+                {
+                    'total_included': 135.0,
+                    'total_excluded': 100.0,
+                    'tax_values_list': (
+                        (100.0, 35.0),
+                    ),
+                },
+                product_values={'volume': 100.0},
+            ),
+            self._prepare_test(
+                "product.volume > 100 and 10 or 5",
+                100.0,
+                {
+                    'total_included': 110.0,
+                    'total_excluded': 100.0,
+                    'tax_values_list': (
+                        (100.0, 10.0),
+                    ),
+                },
+                product_values={'volume': 105.0},
+            ),
+            self._prepare_test(
+                "product.volume > 100 and 10 or 5",
+                100.0,
+                {
+                    'total_included': 105.0,
+                    'total_excluded': 100.0,
+                    'tax_values_list': (
+                        (100.0, 5.0),
+                    ),
+                },
+                product_values={'volume': 50.0},
+            ),
+            self._prepare_test(
+                "product.volume > 100 and 5 or None",
+                100.0,
+                {
+                    'total_included': 100.0,
+                    'total_excluded': 100.0,
+                    'tax_values_list': [],
+                },
+                product_values={'volume': 50.0},
+            ),
+        ]
```

Comment:
We probably have to add more test cases once @tsb-odoo made some tests :sweat_smile: